sw/device/lib/drivers/gpio/gpio.c

```diff
@@ -1,213 +1,624 @@
+            *is_pending = true;
+        else
+            *is_pending = false;
+        return GpioOk;
+    }
+    else
+    {
+        *is_pending = false;
+        return GpioError;
+    }
+}
 
-  gpio->params = params;
+gpio_result_t gpio_intr_check_stat_lvl_low (gpio_pin_number_t pin, bool *is_pending)
+{
+    if (pin >= 0 && pin < MAX_PIN)
+    {
+        if (getBitfield(gpio_peri->INTRPT_LVL_LOW_STATUS0, 1, pin) == 1)
+            *is_pending = true;
+        else
+            *is_pending = false;
+        return GpioOk;
+    }
+    else
+    {
+        *is_pending = false;
+        return GpioError;
+    }
+}
+
+gpio_result_t gpio_intr_check_stat_lvl_high (gpio_pin_number_t pin, bool *is_pending)
+{
+    if (pin >= 0 && pin < MAX_PIN)
+    {
+        if (getBitfield(gpio_peri->INTRPT_LVL_HIGH_STATUS0, 1, pin) == 1)
+            *is_pending = true;
+        else
+            *is_pending = false;
+        return GpioOk;
+    }
+    else
+    {
+        *is_pending = false;
+        return GpioError;
+    }
+}
+
+gpio_result_t gpio_intr_check_stat (gpio_pin_number_t pin, bool *is_pending)
+{
+    if (pin >= 0 && pin < MAX_PIN)
+    {
+        if (getBitfield(gpio_peri->INTRPT_STATUS0, 1, pin) == 1)
+            *is_pending = true;
+        else
+            *is_pending = false;
+        return GpioOk;
+    }
+    else
+    {
+        *is_pending = false;
+        return GpioError;
+    }
+}
+
+gpio_result_t gpio_intr_clear_stat_rise (gpio_pin_number_t pin)
+{
+    if (pin >= 0 && pin < MAX_PIN)
+    {
+        setBitfield(&(gpio_peri->INTRPT_RISE_STATUS0), 1, 1, pin);
+        return GpioOk;
+    }
+    else
+    {
+        return GpioError;
+    }
+}
+
+gpio_result_t gpio_intr_clear_stat_fall (gpio_pin_number_t pin)
+{
+    if (pin >= 0 && pin < MAX_PIN)
+    {
+        setBitfield(&(gpio_peri->INTRPT_FALL_STATUS0), 1, 1, pin);
+        return GpioOk;
+    }
+    else
+    {
+        return GpioError;
+    }
+}
+
+gpio_result_t gpio_intr_clear_stat_lvl_low (gpio_pin_number_t pin)
+{
+    if (pin >= 0 && pin < MAX_PIN)
+    {
+        setBitfield(&(gpio_peri->INTRPT_LVL_LOW_STATUS0), 1, 1, pin);
+        return GpioOk;
+    }
+    else
+    {
+        return GpioError;
+    }
+}
+
+gpio_result_t gpio_intr_clear_stat_lvl_high (gpio_pin_number_t pin)
+{
+    if (pin >= 0 && pin < MAX_PIN)
+    {
+        setBitfield(&(gpio_peri->INTRPT_LVL_HIGH_STATUS0), 1, 1, pin);
+        return GpioOk;
+    }
+    else
+    {
+        return GpioError;
+    }
+}
+
+gpio_result_t gpio_intr_clear_stat (gpio_pin_number_t pin)
+{
+    if (pin >= 0 && pin < MAX_PIN)
+    {
+        setBitfield(&(gpio_peri->INTRPT_STATUS0), 1, 1, pin);
+        return GpioOk;
+    }
+    else
+    {
+        return GpioError;
+    }
+}
 
-  return kGpioOk;
+gpio_result_t gpio_intr_set_mode (bool gpio_intr_mode)
+{
+    if (gpio_intr_mode)
+        setBitfield(&(gpio_peri->CFG), 1, 1, 0);
+    else
+        setBitfield(&(gpio_peri->CFG), 0, 1, 0);
+    return GpioOk;
```

Comment:
here, we are returning always the same regardless of the branch condition, thus, is it truly needed to return anything?